### PR TITLE
Correctly type value accessors & add format/analyze checks to workflow

### PR DESCRIPTION
**Pull request by kuhnroyal**
I am getting a lot of cast errors in custom form fields which don't have a value accessor. This is a problem compared to the pre-NNBD version.
I have added better typings to the internal state valueAccessor and an assert that shows up right away in debug mode.

As an additional bonus I added workflow jobs to check format and analyzer issues.

### CODE REVIEW

File: lib/src/widgets/reactive_form_field.dart [hunk 1]
```diff
@@ -143,8 +143,12 @@ class ReactiveFormFieldState<ModelDataType, ViewDataType>
   /// See [ControlValueAccessor].
   @protected
   @visibleForTesting
-  ControlValueAccessor<dynamic, dynamic> selectValueAccessor() {
-    return DefaultValueAccessor<ModelDataType>();
+  ControlValueAccessor<ModelDataType, ViewDataType> selectValueAccessor() {
+    assert(
+      ModelDataType == ViewDataType,
+      "Must provide a valueAccessor for field $widget as ModelDataType and ViewDataType don't match.",
```
Comment (joanpablo) [inline lib/src/widgets/reactive_form_field.dart:149]: I don't understand very well this assert. Could you explain me? or make a test to see why this is useful?

**Review by joanpablo — CHANGES_REQUESTED** [review]
Hi @kuhnroyal great job, I left you a comment. Thanks

File: lib/src/widgets/reactive_form_field.dart (hunk 1 above)
Comment (joanpablo) [reply lib/src/widgets/reactive_form_field.dart:149]: I understand now, Good Job

Comment (joanpablo) [reply lib/src/widgets/reactive_form_field.dart:149]: But please add a test for this new assert.

Comment (kuhnroyal) [reply lib/src/widgets/reactive_form_field.dart:149]: I will try to get this done tomorrow.

Comment (kuhnroyal) [reply lib/src/widgets/reactive_form_field.dart:149]: Added tests

**Review by joanpablo — APPROVED** [review]

**Pull request merged**